labrad/server.py

```diff
@@ -352,6 +363,17 @@ def startup(self, protocol):
             self.disconnect(e)
             raise
 
+    @property
+    def client(self):
+        if threadable.isInIOThread():
+            # We're in the reactor thread, so can return shared async client.
+            return self.__async_client
+        else:
+            if not hasattr(self.__sync_client, 'cxn'):
+                backend = labrad.backend.TwistedConnection(async_cxn=self.__protocol)
+                self.__sync_client.cxn = Client(backend)
+            return self.__sync_client.cxn
```

Comment:
See the line above, `self.__sync_client.cxn` is an instance of `Client`. This is the synchronous labrad interface we use in command line scripts, etc. Server calls block rather than returning `Deferred`s, which is what `AsyncClient` does. Because `self.__sync_client` is a [threading.local](https://docs.python.org/2/library/threading.html#threading.local) so each thread gets its own `Client` instance, all of them backed by the same underlying twisted connection. The reactor thread on the other hand gets the `AsyncClient` instance, of which there is only one.